File: lib-segment/src/main/java/websiteschema/mpsegment/pinyin/Window.java
package websiteschema.mpsegment.pinyin;

import java.util.ArrayList;
import java.util.List;

public class Window<T> {

    private int size = 0;
    private List<T> datas = new ArrayList<T>();

    public Window(int size) {
        this.size = size;
    }

    public void clear() {
        datas.clear();
    }

    public void add(T data) {
        datas.add(data);
        if (datas.size() > size) {
            datas.remove(0);
        }
    }

    public T[] toArray(T[] t) {
        return datas.toArray(t);
    }
}

File: lib-segment/src/main/scala/me/yingrui/segment/pinyin/Window.scala
package me.yingrui.segment.pinyin

import collection.mutable.ListBuffer

class Window[T](size: Int) {

    private val datas = ListBuffer[T]()

    def clear() {
        datas.clear()
    }

    def add(data: T) {
        datas += (data)
        if (datas.size > size) {
            datas.remove(0)
        }
    }

    def toArray() = datas.toList
}
